### PR TITLE
fix(ui): stage conditions update

### LINKED ISSUE #4708: fix(ui, api): run conditions on stage
add run condition, save. 
reopen run condition -> if the variable is not builtin (as a value in payload), the variable name is empty in the form
add lua condition, save, previous run conditions stay in yml
try to remove conditions, save: not removed.
with edit as yml & ui

### PR DESCRIPTION
**Pull request by yesnault**
close #4708

Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11240.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11242.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11242.1 ✘  
* Stage 1
  * Engine Test with DB ✘

File: engine/api/pipeline/pipeline_stage.go [hunk 1]
```diff
@@ -33,7 +27,7 @@ func LoadStage(db gorp.SqlExecutor, pipelineID int64, stageID int64) (*sdk.Stage
 	var stage sdk.Stage
 	rows, err := db.Query(query, pipelineID, stageID)
 	if err == sql.ErrNoRows {
-		return nil, ErrNoStage
+		return nil, fmt.Errorf("stage does not exist")
```
Comment (fsamin) [inline engine/api/pipeline/pipeline_stage.go]: raise error with stack

File: engine/api/pipeline/pipeline_stage.go [hunk 2]
```diff
@@ -56,21 +50,23 @@ func InsertStage(db gorp.SqlExecutor, s *sdk.Stage) error {
...
-		if err != nil {
-			return sdk.WrapError(err, "cannot marshal condition for stage %d", s.ID)
-		}
-		if _, err := db.Exec(query, conditionsBts, s.ID); err != nil {
-			return err
-		}
+	conditionsBts, err := json.Marshal(s.Conditions)
+
+	if err != nil {
+		return sdk.WrapError(err, "cannot marshal condition for stage %d", s.ID)
```
Comment (fsamin) [inline engine/api/pipeline/pipeline_stage.go]: return error with stack. BTW the wrap message will is useless

File: engine/api/pipeline/pipeline_stage.go [hunk 3]
```diff
@@ -56,21 +50,23 @@ func InsertStage(db gorp.SqlExecutor, s *sdk.Stage) error {
...
+	}
+	if _, err := db.Exec(query, conditionsBts, s.ID); err != nil {
+		return err
```
Comment (fsamin) [inline engine/api/pipeline/pipeline_stage.go]: return error with stack

**Comment by ovh-cds** [comment]
CDS Report it#11278.0 ✘  
* w-it
  * w-it ✘

**Pull request merged**